test/unit/modules/transport.js

```diff
@@ -1028,550 +1220,1161 @@ describe('transport', () => {
+						return expect(
+							__private.receiveSignatures.calledWith(query.signatures)
+						).to.be.true;
+					});
+				});
+				describe('when library.schema.validate fails', () => {
+					var validateErr;
 
-					describe('when __private.receiveSignature fails', () => {
-						it('should call callback with error = null');
+					beforeEach(done => {
+						query = {
+							signatures: [SAMPLE_SIGNATURE_1],
+						};
+						validateErr = new Error('Transaction query did not match schema');
+						validateErr.code = 'INVALID_FORMAT';
 
-						it(
-							'should call callback with result = {success: false, message: err}'
-						);
+						library.schema.validate = sinonSandbox
+							.stub()
+							.callsArgWith(2, validateErr);
+						__private.receiveSignatures = sinonSandbox.stub();
+						transportInstance.shared.postSignatures(query);
+						done();
 					});
 
-					describe('when __private.receiveSignature succeeds', () => {
-						it('should call callback with error = null');
-
-						it('should call callback with result = {success: true}');
+					it('should call library.logger.debug with "Invalid signatures body" and err as arguments', () => {
+						return expect(
+							library.logger.debug.calledWith(
+								'Invalid signatures body',
+								validateErr
+							)
+						).to.be.true;
 					});
 				});
 			});
 
 			describe('getSignatures', () => {
-				it('should call modules.transactions.getMultisignatureTransactionList');
-
-				it(
-					'should call modules.transactions.getMultisignatureTransactionList with true'
-				);
+				var req;
 
-				it(
-					'should call modules.transactions.getMultisignatureTransactionList with constants.maxSharedTxs'
-				);
+				beforeEach(done => {
+					req = {};
+					modules.transactions.getMultisignatureTransactionList = sinonSandbox
+						.stub()
+						.returns(multisignatureTransactionsList);
+					transportInstance.shared.getSignatures(req, (err, res) => {
+						error = err;
+						result = res;
+						done();
+					});
+				});
 
-				describe('for every transaction', () => {
-					describe('when trs.signatures are defined', () => {
-						describe('and trs.signatures.length is defined', () => {
-							describe('signature', () => {
-								it('should assign transaction: trs.id');
+				it('should call modules.transactions.getMultisignatureTransactionList with true and constants.maxSharedTxs', () => {
+					return expect(
+						modules.transactions.getMultisignatureTransactionList.calledWith(
+							true,
+							constants.maxSharedTxs
+						)
+					).to.be.true;
+				});
 
-								it('should assign signatures: trs.signatures');
-							});
-						});
+				describe('when all transactions returned by modules.transactions.getMultisignatureTransactionList are multisignature transactions', () => {
+					it('should call callback with error = null and result = {success: true, signatures: signatures} where signatures contains all transactions', () => {
```

Comment:
Possibly split this test and all following for 2 separate ones - result and error checking. The description will be shorter and tests will be kept more unit.